src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepositoryTest.java

```diff
@@ -0,0 +1,101 @@
+package stashpullrequestbuilder.stashpullrequestbuilder;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import hudson.model.AbstractProject;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import stashpullrequestbuilder.stashpullrequestbuilder.stash.StashApiClient;
+import stashpullrequestbuilder.stashpullrequestbuilder.stash.StashPullRequestResponseValue;
+import stashpullrequestbuilder.stashpullrequestbuilder.stash.StashPullRequestResponseValueRepository;
+import stashpullrequestbuilder.stashpullrequestbuilder.stash.StashPullRequestResponseValueRepositoryRepository;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StashRepositoryTest {
+
+  private StashRepository stashRepository;
+
+  @Mock private StashBuildTrigger trigger;
+  @Mock private StashPullRequestsBuilder builder;
+  @Mock private AbstractProject project;
+  @Mock private StashApiClient stashApiClient;
+  @Mock private StashPullRequestResponseValue pullRequest;
+  @Mock private StashPullRequestResponseValueRepository repository;
+  @Mock private StashPullRequestResponseValueRepositoryRepository repoRepo;
+
+  @Test
+  public void getTargetPullRequestsReturnsEmptyListForNoPullRequests() {
+    stashRepository = new StashRepository(builder, trigger, stashApiClient);
+
+    when(builder.getProject()).thenReturn(project);
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.emptyList());
+
+    Collection<StashPullRequestResponseValue> buildablePullRequests =
+        stashRepository.getTargetPullRequests();
+    assertThat(buildablePullRequests, empty());
+
+    verify(stashApiClient, times(1)).getPullRequests();
+  }
+
+  @Test
+  public void getTargetPullRequestsAcceptsOpenPullRequests() {
+    stashRepository = new StashRepository(builder, trigger, stashApiClient);
+
+    when(builder.getProject()).thenReturn(project);
+    when(stashApiClient.getPullRequests()).thenReturn(Arrays.asList(pullRequest));
+    when(pullRequest.getState()).thenReturn("OPEN");
+    when(trigger.getCiSkipPhrases()).thenReturn("NO TEST");
+    when(pullRequest.getFromRef()).thenReturn(repository);
+    when(pullRequest.getToRef()).thenReturn(repository);
+    when(repository.getRepository()).thenReturn(repoRepo);
+
+    Collection<StashPullRequestResponseValue> buildablePullRequests =
+        stashRepository.getTargetPullRequests();
+    assertThat(buildablePullRequests, hasSize(1));
+    assertThat(buildablePullRequests, hasItems(pullRequest));
```

Comment:
use `contains()` to do the two assertions at once (it asserts there are no more items)

Reply:
Done. Thank you!